src/hugr.rs

```diff
@@ -134,6 +124,23 @@ impl NodeType {
         self.input_extensions.as_ref()
     }
 
+    /// The input and output extensions for this node, if set.
+    ///
+    /// `None`` if the [Self::input_extensions] is `None`.
+    /// Otherwise, will return Some, with the output extensions computed from the node's delta
+    pub fn io_extensions(&self) -> Option<(&ExtensionSet, ExtensionSet)> {
+        self.input_extensions.as_ref().map(|e| {
+            (
+                e,
+                self.op
+                    .dataflow_signature()
+                    .map(|ft| ft.extension_reqs)
+                    .unwrap_or_default()
```

Comment:
This is a change to the semantics introduced in #680 but I believe closer to what we had before - where a non-dataflow node would return `FunctionType::default()` (rather than `None` in #680) which meant the empty set of `extension_reqs`